contracts/locked-astroport-vault/src/state.rs

```diff
@@ -188,3 +320,128 @@ pub struct StateResponse {
+    fn fee_config_recipients_must_sum_to_one() {
+        let deps = mock_dependencies();
+
+        let fee_config = super::FeeConfig {
+            fee_rate: Decimal::percent(1),
+            fee_recipients: vec![
+                ("addr1".to_string(), Decimal::percent(20)),
+                ("addr2".to_string(), Decimal::percent(50)),
+            ],
+        };
+        assert!(fee_config
+            .check(&deps.as_ref())
+            .unwrap_err()
+            .to_string()
+            .contains("Sum of fee recipient percentages must be 100%"));
+    }
+
+    #[test]
+    fn fee_config_recipient_weights_must_be_greater_than_zero() {
+        let deps = mock_dependencies();
+
+        let fee_config = super::FeeConfig {
+            fee_rate: Decimal::percent(1),
+            fee_recipients: vec![
+                ("addr1".to_string(), Decimal::percent(100)),
+                ("addr2".to_string(), Decimal::zero()),
+            ],
+        };
+        assert!(fee_config
+            .check(&deps.as_ref())
+            .unwrap_err()
+            .to_string()
+            .contains("Fee recipient percentages must be greater than zero"));
+    }
+
+    #[test]
+    fn fee_msgs_from_asset_works() {
+        let env = mock_env();
+
+        let fee_config = super::FeeConfig {
+            fee_rate: Decimal::percent(1),
+            fee_recipients: vec![(Addr::unchecked("addr1"), Decimal::percent(100))],
+        };
+        let asset = Asset::new(AssetInfo::native("uusdc"), 100u128);
+        let (msgs, asset_after_fee) = fee_config.fee_msgs_from_asset(asset, &env).unwrap();
+        assert_eq!(msgs.len(), 1);
+        assert_eq!(asset_after_fee.amount, Uint128::new(99));
+    }
+
+    #[test]
+    fn fee_msgs_from_asset_works_with_zero_fee_rate() {
+        let env = mock_env();
+
+        let fee_config = super::FeeConfig {
+            fee_rate: Decimal::zero(),
+            fee_recipients: vec![],
+        };
+        let asset = Asset::new(AssetInfo::native("uusdc"), 100u128);
+        let (msgs, asset_after_fee) = fee_config.fee_msgs_from_asset(asset, &env).unwrap();
+        assert_eq!(msgs.len(), 0);
+        assert_eq!(asset_after_fee.amount, Uint128::new(100));
+    }
+
+    #[test]
+    fn fee_msgs_from_assets_works() {
+        let env = mock_env();
+
+        let fee_config = super::FeeConfig {
+            fee_rate: Decimal::percent(1),
+            fee_recipients: vec![(Addr::unchecked("addr1"), Decimal::percent(100))],
+        };
+        let assets = vec![
+            Asset::new(AssetInfo::native("uusdc"), 100u128),
+            Asset::new(AssetInfo::native("uatom"), 200u128),
+        ]
+        .into();
+        let (msgs, assets_after_fee) = fee_config.fee_msgs_from_assets(&assets, &env).unwrap();
+        println!("{:?}", msgs);
+        assert_eq!(msgs.len(), 2);
+        assert_eq!(assets_after_fee.to_vec()[0].amount, Uint128::new(99));
+        assert_eq!(assets_after_fee.to_vec()[1].amount, Uint128::new(198));
+    }
```

Comment:
add unit test case for this case but with multiple recipients

Reply:
done